--- cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/ResourceEventArgs.pypp.cpp.orig	2013-05-26 17:58:46.000000000 +0200
+++ cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/ResourceEventArgs.pypp.cpp	2015-11-21 16:33:29.302275302 +0100
@@ -4,6 +4,14 @@
 #include "generators/include/python_CEGUI.h"
 #include "ResourceEventArgs.pypp.hpp"
 
+namespace boost { namespace python {
+template <class D>
+inline object make_setter(D const& x)
+{
+    return detail::make_setter(x, default_call_policies(), is_member_pointer<D>(), 0);
+}
+}}
+
 namespace bp = boost::python;
 
 void register_ResourceEventArgs_class(){
--- cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/Scheme.pypp.cpp.orig	2013-05-26 17:58:46.000000000 +0200
+++ cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/Scheme.pypp.cpp	2015-11-21 16:26:43.769253262 +0100
@@ -4,6 +4,14 @@
 #include "generators/include/python_CEGUI.h"
 #include "Scheme.pypp.hpp"
 
+namespace boost { namespace python {
+template <class D>
+inline object make_setter(D const& x)
+{
+    return detail::make_setter(x, default_call_policies(), is_member_pointer<D>(), 0);
+}
+}}
+
 namespace bp = boost::python;
 
 void register_Scheme_class(){
--- cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/StdPairCEGUIStringString.pypp.cpp.orig	2013-05-26 17:58:46.000000000 +0200
+++ cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/StdPairCEGUIStringString.pypp.cpp	2015-11-21 16:24:44.334020922 +0100
@@ -4,6 +4,14 @@
 #include "generators/include/python_CEGUI.h"
 #include "StdPairCEGUIStringString.pypp.hpp"
 
+namespace boost { namespace python {
+template <class D>
+inline object make_setter(D const& x)
+{
+    return detail::make_setter(x, default_call_policies(), is_member_pointer<D>(), 0);
+}
+}}
+
 namespace bp = boost::python;
 
 void register_StdPairCEGUIStringString_class(){
--- cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/WindowFactoryManager.pypp.cpp.orig	2013-05-26 17:58:46.000000000 +0200
+++ cegui-0.8.4/cegui/src/ScriptModules/Python/bindings/output/CEGUI/WindowFactoryManager.pypp.cpp	2015-11-21 16:35:45.367135886 +0100
@@ -4,6 +4,14 @@
 #include "generators/include/python_CEGUI.h"
 #include "WindowFactoryManager.pypp.hpp"
 
+namespace boost { namespace python {
+template <class D>
+inline object make_setter(D const& x)
+{
+    return detail::make_setter(x, default_call_policies(), is_member_pointer<D>(), 0);
+}
+}}
+
 namespace bp = boost::python;
 
 void register_WindowFactoryManager_class(){
