--- CEGUI-0.5.0/acinclude.m4.orig	2006-10-25 12:54:43.000000000 +0200
+++ CEGUI-0.5.0/acinclude.m4	2007-05-27 18:37:36.052672449 +0200
@@ -55,9 +55,6 @@
     CEGUI_CHECK_XERCES(xerces, [cegui_found_xerces=yes], [cegui_found_xerces=no])
     CEGUI_CHECK_EXPAT(expat, [cegui_found_expat=yes], [cegui_found_expat=no])
 
-    dnl save lots of linker aggro ;)
-    LIBS="$xerces_LIBS $LIBS"
-
     dnl Find out what user disabled as far as parsers go
     AC_ARG_ENABLE([xerces-c], AC_HELP_STRING([--disable-xerces-c], [Disables building of the Xerces-C++ XML parser module.]),
                 [cegui_with_xerces=$enableval], [cegui_with_xerces=yes])
@@ -267,7 +264,7 @@
     [cegui_default_image_codec=$withval], [cegui_default_image_codec=none])
     AC_PATH_XTRA
     cegui_saved_LIBS="$LIBS"
-    LIBS="$X_PRE_LIBS $X_LIBS $X_EXTRA_LIBS"
+    LIBS="$X_LIBS $X_EXTRA_LIBS"
 
     AC_SEARCH_LIBS(glInterleavedArrays, MesaGL GL, cegui_found_lib_GL=yes, cegui_found_lib_GL=no)
     AC_SEARCH_LIBS(gluOrtho2D, MesaGLU GLU, cegui_found_lib_GLU=yes,  cegui_found_lib_GLU=no)
@@ -309,7 +306,7 @@
         dnl FreeImage 
         if test x$cegui_with_freeimage = xyes ; then 
             AC_LANG_PUSH(C++)
-            AC_SEARCH_LIBS(FreeImage_GetVersion, freeimage, [cegui_with_freeimage_lib=yes], [cegui_with_freeimage_lib=no])
+            AC_CHECK_LIB(freeimage, FreeImage_GetVersion, [cegui_with_freeimage_lib=yes], [cegui_with_freeimage_lib=no])
             AC_LANG_POP(C++)
             AC_CHECK_HEADER(FreeImage.h, [cegui_with_freeimage_header=yes], [cegui_with_freeimage_header=no], [])
             if test x$cegui_with_freeimage_lib = xyes -a x$cegui_with_freeimage_header = xyes ; then
@@ -537,7 +534,7 @@
 ])
 
 AC_DEFUN([CEGUI_CHECK_LUA],[
-    PKG_CHECK_MODULES(Lua, lua >= 5.0 lua < 5.1, [cegui_found_lua=yes], [cegui_found_lua=no])
+    PKG_CHECK_MODULES(Lua, lua50 >= 5.0 lua50 < 5.1, [cegui_found_lua=yes], [cegui_found_lua=no])
 
     AC_ARG_ENABLE([lua-module], AC_HELP_STRING([--disable-lua-module], [Disables building of the Lua scripting module.]),
                 [cegui_with_lua=$enableval], [cegui_with_lua=yes])
--- CEGUI-0.5.0/RendererModules/OpenGLGUIRenderer/Makefile.am.orig	2006-07-28 03:24:36.000000000 +0200
+++ CEGUI-0.5.0/RendererModules/OpenGLGUIRenderer/Makefile.am	2007-05-27 18:49:01.872325129 +0200
@@ -3,6 +3,7 @@
 pkginclude_HEADERS = openglrenderer.h opengltexture.h
 lib_LTLIBRARIES=libCEGUIOpenGLRenderer.la
 libCEGUIOpenGLRenderer_la_LDFLAGS = -version-info @CEGUI_OPENGL_VERSION_INFO@
+libCEGUIOpenGLRenderer_la_LIBADD = $(top_builddir)/src/libCEGUIBase.la -lGLU -lGL
 libCEGUIOpenGLRenderer_la_SOURCES = openglrenderer.cpp opengltexture.cpp
 
 
--- CEGUI-0.5.0/ScriptingModules/CEGUILua/tolua++bin/Makefile.am.orig	2006-10-17 13:58:04.000000000 +0200
+++ CEGUI-0.5.0/ScriptingModules/CEGUILua/tolua++bin/Makefile.am	2007-05-27 18:59:43.330163303 +0200
@@ -1,4 +1,5 @@
-INCLUDES = -I$(top_srcdir)/ScriptingModules/CEGUILua/tolua++
+INCLUDES = -I$(top_srcdir)/ScriptingModules/CEGUILua/tolua++ \
+	   $(Lua_CFLAGS)
 
 bin_PROGRAMS = tolua++cegui
 tolua__cegui_SOURCES = tolua.c toluabind.c toluabind.h
--- CEGUI-0.5.0/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am.orig	2006-07-25 10:34:51.000000000 +0200
+++ CEGUI-0.5.0/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am	2007-05-27 19:40:54.206484382 +0200
@@ -1,9 +1,12 @@
 INCLUDES = -I$(top_srcdir)/include \
            -I$(top_srcdir)/ScriptingModules/CEGUILua/tolua++ \
-           -I$(top_srcdir)/ScriptingModules/CEGUILua/LuaScriptModule/include
+           -I$(top_srcdir)/ScriptingModules/CEGUILua/LuaScriptModule/include \
+	   $(Lua_CFLAGS)
 lib_LTLIBRARIES = libCEGUILuaScriptModule.la
 
 libCEGUILuaScriptModule_la_LDFLAGS = -version-info @CEGUI_VERSION_INFO@
+libCEGUILuaScriptModule_la_LIBADD = $(top_builddir)/src/libCEGUIBase.la \
+				    $(top_builddir)/ScriptingModules/CEGUILua/tolua++/libCEGUItoluapp.la
 
 libCEGUILuaScriptModule_la_SOURCES = lua_CEGUI.cpp \
                                      CEGUILua.cpp \
