--- CEGUI-0.6.1/acinclude.m4.orig	2008-06-07 18:04:08.000000000 +0200
+++ CEGUI-0.6.1/acinclude.m4	2008-08-29 21:31:32.846436662 +0200
@@ -67,9 +67,6 @@
         cegui_found_tinyxml=no
     fi
 
-    dnl save lots of linker aggro ;)
-    LIBS="$xerces_LIBS $LIBS"
-
     dnl Find out what user disabled as far as parsers go
     AC_ARG_ENABLE([xerces-c], AC_HELP_STRING([--disable-xerces-c], [Disables building of the Xerces-C++ XML parser module.]),
                 [cegui_with_xerces=$enableval], [cegui_with_xerces=yes])
@@ -406,7 +403,7 @@
         ;;
     * )
         AC_PATH_XTRA
-        LIBS="$X_PRE_LIBS $X_LIBS $X_EXTRA_LIBS"
+        LIBS="$X_LIBS $X_EXTRA_LIBS"
 
         AC_SEARCH_LIBS(glInterleavedArrays, MesaGL GL, cegui_found_lib_GL=yes, cegui_found_lib_GL=no)
         AC_SEARCH_LIBS(gluOrtho2D, MesaGLU GLU, cegui_found_lib_GLU=yes,  cegui_found_lib_GLU=no)
@@ -500,7 +500,7 @@
             if test "x$prefix" != xNONE ; then
                 PATH="$prefix/bin:$prefix/usr/bin:$PATH"
             fi
-            AC_PATH_PROG(cegui_corona_config, corona-config, no, [PATH])
+            AC_PATH_PROG(cegui_corona_config, corona-config, no, [$PATH])
             AC_MSG_CHECKING([for corona - version >= 1.2.0])
             cegui_with_corona=no
             if test "$cegui_corona_config" = "no" ; then
@@ -702,7 +699,7 @@
 
     dnl If that did not work, try again with an alternate name for the packages (as used on (K)Ubuntu etc)
     if test x$cegui_found_lua = xno; then
-        PKG_CHECK_MODULES(Lua, lua5.1,
+        PKG_CHECK_MODULES(Lua, lua51,
                         [cegui_found_lua=yes; Lua_CFLAGS="$Lua_CFLAGS -DCEGUI_LUA_VER=51"],
                         [PKG_CHECK_MODULES(Lua, lua50,
                                             [cegui_found_lua=yes; Lua_CFLAGS="$Lua_CFLAGS -DCEGUI_LUA_VER=50"],
