--- cegui-0.8.4/CMakeLists.txt.orig	2014-07-07 10:46:25.000000000 +0200
+++ cegui-0.8.4/CMakeLists.txt	2014-10-26 07:04:55.574145118 +0100
@@ -106,7 +106,7 @@
 find_package(Lua51)
 find_package(TOLUAPP)
 find_package(PythonInterp)
-find_package(PythonLibs)
+find_package(PythonLibs ${PYTHON_VERSION_STRING})
 find_package(Boost 1.36.0 COMPONENTS python unit_test_framework system)
 
 find_package(Doxygen)
