--- CEGUI-0.5.0/acinclude.m4.orig	2007-12-26 13:53:05.799672000 +0100
+++ CEGUI-0.5.0/acinclude.m4	2007-12-26 14:06:00.363738552 +0100
@@ -214,6 +214,7 @@
     cegui_saved_CFLAGS="$CPPFLAGS"
     cegui_saved_LIBS="$LIBS"
     
+    AC_LANG_PUSH([C++])
     for cegui_path in $cegui_inc_paths; do
         ifelse($cegui_path, [.], CPPFLAGS="$cegui_saved_CFLAGS", CPPFLAGS="-I$cegui_path $cegui_saved_CFLAGS")
         AC_PREPROC_IFELSE([#include <irrlicht.h>], [cegui_irr_h_found=yes; cegui_irr_flags="$cegui_path"; break],[cegui_irr_h_found=no])
@@ -241,6 +242,7 @@
     else
         ifelse([$3], [], :, [$3])
     fi    
+    AC_LANG_POP([C++])
 ])
 
 AC_DEFUN([CEGUI_ENABLE_OPENGL_RENDERER], [
--- CEGUI-0.5.0/RendererModules/IrrlichtRenderer/IrrlichtMemoryFile.h.orig	2006-06-08 18:25:39.000000000 +0200
+++ CEGUI-0.5.0/RendererModules/IrrlichtRenderer/IrrlichtMemoryFile.h	2007-12-26 17:51:27.802141527 +0100
@@ -43,11 +43,11 @@
 public:
     IrrlichtMemoryFile(const String& filename, const unsigned char* memory, uint32 size);
     virtual ~IrrlichtMemoryFile() {};
-    virtual irr::s32 read(void* buffer, irr::s32 sizeToRead);
-    virtual bool seek(irr::s32 finalPos, bool relativeMovement = false);
-    virtual irr::s32 getSize();
-    virtual irr::s32 getPos();
-    virtual const irr::c8* getFileName();
+    virtual irr::s32 read(void* buffer, irr::u32 sizeToRead);
+    virtual bool seek(long finalPos, bool relativeMovement = false);
+    virtual long getSize() const;
+    virtual long getPos() const;
+    virtual const irr::c8* getFileName() const;
 
 protected:
     String d_filename;
--- CEGUI-0.5.0/RendererModules/IrrlichtRenderer/IrrlichtMemoryFile.cpp.orig	2006-06-08 18:25:39.000000000 +0200
+++ CEGUI-0.5.0/RendererModules/IrrlichtRenderer/IrrlichtMemoryFile.cpp	2007-12-26 14:20:45.676239376 +0100
@@ -39,7 +39,7 @@ IrrlichtMemoryFile::IrrlichtMemoryFile(c
 {
 }
 
-irr::s32 IrrlichtMemoryFile::read(void* buffer, irr::s32 sizeToRead)
+irr::s32 IrrlichtMemoryFile::read(void* buffer, irr::u32 sizeToRead)
 {
     uint32 realReadSize =
         ((d_position + sizeToRead) > d_size) ? d_size - d_position : sizeToRead;
@@ -50,7 +50,7 @@ irr::s32 IrrlichtMemoryFile::read(void* 
     return realReadSize;
 }
 
-bool IrrlichtMemoryFile::seek(irr::s32 finalPos, bool relativeMovement)
+bool IrrlichtMemoryFile::seek(long finalPos, bool relativeMovement)
 {
     uint32 targetPosition = relativeMovement ? d_position : 0;
     targetPosition += finalPos;
@@ -66,17 +66,17 @@ bool IrrlichtMemoryFile::seek(long f
     }
 }
 
-irr::s32 IrrlichtMemoryFile::getSize()
+long IrrlichtMemoryFile::getSize() const
 {
     return d_size;
 }
 
-irr::s32 IrrlichtMemoryFile::getPos()
+long IrrlichtMemoryFile::getPos() const
 {
     return d_position;
 }
 
-const irr::c8* IrrlichtMemoryFile::getFileName()
+const irr::c8* IrrlichtMemoryFile::getFileName() const
 {
     return d_filename.c_str();
 }
--- CEGUI-0.5.0/RendererModules/IrrlichtRenderer/irrlichtrenderer.h.orig	2006-06-08 18:25:39.000000000 +0200
+++ CEGUI-0.5.0/RendererModules/IrrlichtRenderer/irrlichtrenderer.h	2007-12-26 18:00:20.725801530 +0100
@@ -81,7 +81,7 @@ namespace CEGUI
 		virtual ResourceProvider* createResourceProvider(void);
 
 		/*! forward event to CEGUI system */
-		bool OnEvent(irr::SEvent& event);
+		bool OnEvent(const irr::SEvent& event);
 
 
 		/*************************************************************************
--- CEGUI-0.5.0/RendererModules/IrrlichtRenderer/irrlichtrenderer.cpp.orig	2006-08-06 12:49:02.000000000 +0200
+++ CEGUI-0.5.0/RendererModules/IrrlichtRenderer/irrlichtrenderer.cpp	2007-12-26 18:00:36.286589724 +0100
@@ -265,7 +265,7 @@ namespace CEGUI
 	}
 
 	/*! forward event to CEGUI system */
-	bool IrrlichtRenderer::OnEvent(irr::SEvent& event)
+	bool IrrlichtRenderer::OnEvent(const irr::SEvent& event)
 	{
 		return eventpusher->OnEvent(event);
 	}
--- CEGUI-0.5.0/RendererModules/IrrlichtRenderer/IrrlichtEventPusher.h.orig	2006-08-02 20:55:09.000000000 +0200
+++ CEGUI-0.5.0/RendererModules/IrrlichtRenderer/IrrlichtEventPusher.h	2007-12-26 18:01:42.862671474 +0100
@@ -56,7 +56,7 @@ namespace CEGUI
 
 		virtual ~EventPusher(){};
 
-		bool OnEvent(SEvent& event)
+		bool OnEvent(const SEvent& event)
 		{
 			switch(event.EventType)
 			{
--- CEGUI-0.5.0/Samples/common/src/CEGuiIrrlichtBaseApplication.cpp.orig	2006-08-09 21:39:16.000000000 +0200
+++ CEGUI-0.5.0/Samples/common/src/CEGuiIrrlichtBaseApplication.cpp	2007-12-26 17:59:02.811854993 +0100
@@ -140,7 +140,7 @@ void CEGuiIrrlichtBaseApplication::clean
     // Nothing to do here.
 }
 
-bool CEGuiIrrlichtBaseApplication::OnEvent(irr::SEvent event)
+bool CEGuiIrrlichtBaseApplication::OnEvent(const irr::SEvent &event)
 {
     // cegui samples always quit on escape
     if (event.EventType == irr::EET_KEY_INPUT_EVENT)
--- CEGUI-0.5.0/Samples/common/include/CEGuiIrrlichtBaseApplication.h.orig	2006-08-02 20:55:09.000000000 +0200
+++ CEGUI-0.5.0/Samples/common/include/CEGuiIrrlichtBaseApplication.h	2007-12-26 17:58:28.983474838 +0100
@@ -52,7 +52,7 @@ public:
     void    cleanup();
 
     // irrlicht event listener
-    bool OnEvent(irr::SEvent event);
+    bool OnEvent(const irr::SEvent& event);
 
 protected:
     irr::IrrlichtDevice*       d_device;
