diff -urNp -x '*.orig' cegui-0.8.7.org/cegui/src/RendererModules/Ogre/CMakeLists.txt cegui-0.8.7/cegui/src/RendererModules/Ogre/CMakeLists.txt
--- cegui-0.8.7.org/cegui/src/RendererModules/Ogre/CMakeLists.txt	2016-04-23 20:18:29.000000000 +0200
+++ cegui-0.8.7/cegui/src/RendererModules/Ogre/CMakeLists.txt	2022-03-11 21:23:25.595133918 +0100
@@ -7,5 +7,5 @@ cegui_gather_files()
 cegui_add_library(${CEGUI_TARGET_NAME} CORE_SOURCE_FILES CORE_HEADER_FILES)
 cegui_add_dependency(${CEGUI_TARGET_NAME} OGRE PUBLIC TRUE)
 
-cegui_target_link_libraries(${CEGUI_TARGET_NAME} ${CEGUI_BASE_LIBNAME})
+cegui_target_link_libraries(${CEGUI_TARGET_NAME} ${CEGUI_BASE_LIBNAME} -lpthread)
 
